### PR TITLE
Adds `swayfmt` README/CONTRIBUTING

### PR DESCRIPTION
**Pull request by eureka-cpu**
I figure as this grows we will need some sort of guideline for contributors and testers so I whipped this up

### CODE REVIEW

File: sway-fmt-v2/CONTRIBUTING.md
Comment (kayagokalp) [inline sway-fmt-v2/CONTRIBUTING.md:23]: `sway-fmt-v2` is currently used by `forc-fmt-v2` and we are still distributing `forc-fmt` (which uses `sway-fmt`) with `fuelup`. So our users will have `forc-fmt` installed which can create a confusion here. Considering this is the readme for `sway-fmt-v2` we may want to consider changing this to `forc-fmt-v2` until we make the complete transition and replace/remove v1 from distribution.

**Review by kayagokalp — COMMENTED** [review]
Looks cool, just left a single nit!

File: sway-fmt-v2/CONTRIBUTING.md
Comment (Braqzen) [inline sway-fmt-v2/CONTRIBUTING.md:19]: When I read the points sequentially I wonder about point 3 and what it means to build some manifest and move some result to the specified directory. I think that should be elaborated by specifically mentioning the files i.e. what the manifest file is and what the result file is.

The subsequent section is a code block containing the steps.

I would prefer the following structure

1. Bullet Point 1

  ```bash
  $ command to move to dir
  ```

2. Bullet point 2

  ```bash
  $ clone command
  ```

etc.

I think this is much clearer to understand than multiple points and then 1 code block containing multiple steps e.g. bullet point 1 is omitted but bullet point 3 contains 2 commands; the last two lines of the block. As a beginner I might not be able to distinguish between the steps therefore separating into the aforemonetioned format makes it easier to see which command is performing which action

File: sway-fmt-v2/CONTRIBUTING.md
Comment (Braqzen) [inline sway-fmt-v2/CONTRIBUTING.md:23]: This is not guarenteed to work because it is dependent upon the location from which the command is executed. I think the doc should outline where the command can be run e.g. can it be run inside the `src` directory, can it be run on the same level as the `src`, what happens if you run it outside of that project? A beginner might not know and I sometimes run the command from an incorrect location and an error pops up telling me that the manifest file cannot be located - which is not a great error.

File: sway-fmt-v2/CONTRIBUTING.md
Comment (Braqzen) [inline sway-fmt-v2/CONTRIBUTING.md:36]:
```suggestion
- If there is a bug that isn't represented by a ticket please provide the following details:
```

File: sway-fmt-v2/CONTRIBUTING.md
Comment (Braqzen) [inline sway-fmt-v2/CONTRIBUTING.md:39]: A user may not know how to check the version of any of these tools. Similar to the formatting in a comment above I think this should outline which tool is going to be checked and how to run the corresponding command in order to find the version. Moreover, it should give an example of the output so that the user knows exactly what to copy and paste into the issue because they may copy only some of the information.

It may be helpful to take a look at a visual example of what Sway Apps is doing in terms of its [issue templates](https://github.com/FuelLabs/sway-applications/blob/master/.github/ISSUE_TEMPLATE/bug_report.yml). If you scroll down you'll see it ask for the version, the command to run and it provides an example output

File: sway-fmt-v2/CONTRIBUTING.md
Comment (Braqzen) [inline sway-fmt-v2/CONTRIBUTING.md:43]: I think it would be beneficial to indicate what the expectation of the output format is. Is this a screen shot or a copy and pasted line from the terminal? Does it contain all of the information and if it's copied then has it been formatted in the issue to look like code rather than malformed tokens with uneven indentation etc.? Similarly, an example, possibly of a screen shot and separately a code block, would be beneficial.

File: sway-fmt-v2/CONTRIBUTING.md
Comment (Braqzen) [inline sway-fmt-v2/CONTRIBUTING.md:45]:
```suggestion
- If a feature you would like to see implemented isn't represented by a current ticket please provide the following details:
```

Comment (Braqzen) [reply sway-fmt-v2/CONTRIBUTING.md:45]: cc @simonr0204 :)

File: sway-fmt-v2/CONTRIBUTING.md
Comment (Braqzen) [inline sway-fmt-v2/CONTRIBUTING.md:47]:
```suggestion
  - an example of what the code looks like before being formatted and what the code would look like after formatting
```

File: sway-fmt-v2/CONTRIBUTING.md
Comment (Braqzen) [inline sway-fmt-v2/CONTRIBUTING.md:48]: If a solution is implied to be a change to some code then I think it should be stated because it's not immediately clear to me what a solution could be here

File: sway-fmt-v2/CONTRIBUTING.md
Comment (Braqzen) [inline sway-fmt-v2/CONTRIBUTING.md:49]:
```suggestion

> **Note**: please keep in mind, it is not the goal of `swayfmt` to be identically configurable to `rustfmt` and that some requested features may remain unimplemented until a consensus is reached on adding them
```

File: sway-fmt-v2/CONTRIBUTING.md
Comment (Braqzen) [inline sway-fmt-v2/CONTRIBUTING.md:53]: I think this should point the reader in the direction of how to do that in case they do not know what this means or how to do it. [Here](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) is the link

File: sway-fmt-v2/CONTRIBUTING.md
Comment (Braqzen) [inline sway-fmt-v2/CONTRIBUTING.md:54]:
```suggestion
- If not enough context is provided by an issue please request a more detailed explanation or look to `rustfmt` as a definitive reference
```

File: sway-fmt-v2/CONTRIBUTING.md
Comment (eureka-cpu) [reply sway-fmt-v2/CONTRIBUTING.md:23]: Since it's getting close to being switched over I've left it like this, but I can add a line that is explicit about that too

File: sway-fmt-v2/CONTRIBUTING.md
Comment (Braqzen) [inline sway-fmt-v2/CONTRIBUTING.md:55]:
```suggestion
- When taking on a task consider assigning yourself to the issue or dropping a comment so that a contributor can assign you to the task. This prevents from multiple people taking on the same work
- If the issue consists of a large change then perhaps multiple smaller pull requests should be made while referencing the original issue
- Any changes are encouraged to be opened as a draft pull request until they are ready for review. This makes it easier to track the progress of the issue and allows others to communicate their ideas and suggestions as the work is coming in. Once the work is ready for official review then the pull request should be changed from a draft to an open pull request
```

File: sway-fmt-v2/CONTRIBUTING.md
Comment (Braqzen) [inline sway-fmt-v2/CONTRIBUTING.md:56]: Perhaps a broader comment about CI should be added here so that they know that all of the CI checks must pass?

**Review by Braqzen — COMMENTED** [review]
Lot's of suggestions and comments :)

**Comment by eureka-cpu** [comment]
> Lot's of suggestions and comments :)

Awesome thanks! I'll get on that asap

File: sway-fmt-v2/CONTRIBUTING.md
Comment (sezna) [reply sway-fmt-v2/CONTRIBUTING.md:9]: prerequisite or pre-requisite? I thought it was the former.

File: sway-fmt-v2/CONTRIBUTING.md
Comment (eureka-cpu) [reply sway-fmt-v2/CONTRIBUTING.md]: Gotcha, I was actually using my memory of that to write this, though I should have just referenced it directly

Comment (eureka-cpu) [reply sway-fmt-v2/CONTRIBUTING.md]: I would like to create a template, though that may end up being tricky since it isn't in its own repo

File: sway-fmt-v2/CONTRIBUTING.md
Comment (eureka-cpu) [reply sway-fmt-v2/CONTRIBUTING.md]: I think that was a leftover from copy pasta

File: sway-fmt-v2/CONTRIBUTING.md
Comment (eureka-cpu) [reply sway-fmt-v2/CONTRIBUTING.md]: Unfortunately, outside contributors can't assign themselves or be assigned to my knowledge hence why I have it like this. 😓

File: swayfmt/CONTRIBUTING.md
Comment (Braqzen) [inline swayfmt/CONTRIBUTING.md:17]: This seems to be 2 commands on 1 line. Should there be a `&&` after the `.toml` and before the `mv`?

File: swayfmt/CONTRIBUTING.md
Comment (Braqzen) [inline swayfmt/CONTRIBUTING.md:22]: Does it have to be in the `src`? Pretty sure the current formatter allows you to run it from outside of `src` too.

File: swayfmt/CONTRIBUTING.md
Comment (eureka-cpu) [reply swayfmt/CONTRIBUTING.md:17]: Yes, thank you!

File: swayfmt/CONTRIBUTING.md
Comment (eureka-cpu) [reply swayfmt/CONTRIBUTING.md:22]: Ah, right you are, as long as there is a `Forc.toml` it should work

**Review by Braqzen — COMMENTED** [review]
1. I think the contributing doc should contain info about contributing guidelines and expectations rather than installation and removal instructions. Those instructions should be in the readme.
2. When I use the `where` command I get `command not found`. This seems to be a specific command to some OS (perhaps MAC?). On linux, and there's a lot of distro's so I don't know if how many it applies to, there is a command `which` which presents the path similar to the example output.
3. The testing section is only really saying how to run the plugin. I think showing the user how to run your project should also be under a section in the readme while an actual "testing" section i.e. "create a code example with bad formatting and compare it to expected formatting after your change" should be explained in the contributing doc.
    - How do you structure your tests etc.

As a summary, the README should contain info on how to install, uninstall and run the project while the contributing should explain the expectations for code quality and proving that the change works.
P.S `where` is not a universal command

**Comment by eureka-cpu** [comment]
> 1. I think the contributing doc should contain info about contributing guidelines and expectations rather than installation and removal instructions. Those instructions should be in the readme.
> 
>     2. When I use the `where` command I get `command not found`. This seems to be a specific command to some OS (perhaps MAC?). On linux, and there's a lot of distro's so I don't know if how many it applies to, there is a command `which` which presents the path similar to the example output.
> 
>     3. The testing section is only really saying how to run the plugin. I think showing the user how to run your project should also be under a section in the readme while an actual "testing" section i.e. "create a code example with bad formatting and compare it to expected formatting after your change" should be explained in the contributing doc.
>        
>        * How do you structure your tests etc.
> 
> 
> As a summary, the README should contain info on how to install, uninstall and run the project while the contributing should explain the expectations for code quality and proving that the change works. P.S `where` is not a universal command

So I do have the installation instructions set in the readme to direct the user to `fuelup` since that's the way anyone not contributing _should_ install `swayfmt/forc-fmt` and anyone who would be contributing might want to install it from source.

File: swayfmt/CONTRIBUTING.md
Comment (sezna) [inline swayfmt/CONTRIBUTING.md:45]:
```suggestion
Move into a sway project directory, and execute the binary:
```

File: swayfmt/CONTRIBUTING.md
Comment (sezna) [inline swayfmt/CONTRIBUTING.md:37]:
```suggestion
Move into a sway project directory, and execute the binary:
```

File: swayfmt/CONTRIBUTING.md
Comment (sezna) [inline swayfmt/CONTRIBUTING.md:58]:
```suggestion
- If there is a bug that isn't represented by an issue please provide the following details:
```

File: swayfmt/CONTRIBUTING.md
Comment (sezna) [inline swayfmt/CONTRIBUTING.md:66]: we could just use an [issue template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository) for this

File: swayfmt/CONTRIBUTING.md
Comment (sezna) [inline swayfmt/CONTRIBUTING.md:78]: I'm not sure we need to write this line, as we have CI and github approval settings encoded in the PR process. Writing it here means we need to keep this document in sync with our policies, thus introducing the possibility of drift 
```suggestion
```

**Review by sezna — COMMENTED** [review]
nice! thanks for putting this together. I'd also encourage you to look at other contributing guidelines from Fuel and trying to reuse what you can. There's a lot of work that can be saved here by referencing our [code standards](https://github.com/FuelLabs/rfcs/tree/master/text/code-standards) instead of spelling that out here.

1. [fuel core contributing document](https://github.com/FuelLabs/fuel-core/blob/master/CONTRIBUTING.md)
2. [app dev repo](https://github.com/FuelLabs/sway-applications/blob/master/CONTRIBUTING.md)

File: swayfmt/CONTRIBUTING.md
Comment (eureka-cpu) [reply swayfmt/CONTRIBUTING.md]: #2725

**Review by Braqzen — APPROVED** [review]
Only thing that I would change is that the contributing installation first tells the reader to change to any directory they want and then clone the repo into that directory however the last line, which builds, points at the repo at the top of their path rather than inside the directory which they have moved to. If they follow the instructions directly then they'll get an error

**Comment by eureka-cpu** [comment]
> Only thing that I would change is that the contributing installation first tells the reader to change to any directory they want and then clone the repo into that directory however the last line, which builds, points at the repo at the top of their path rather than inside the directory which they have moved to. If they follow the instructions directly then they'll get an error

the directory they choose could have any name, what would you recommend? I could just leave out the bit about choosing a directory

**Comment by Braqzen** [comment]
> > Only thing that I would change is that the contributing installation first tells the reader to change to any directory they want and then clone the repo into that directory however the last line, which builds, points at the repo at the top of their path rather than inside the directory which they have moved to. If they follow the instructions directly then they'll get an error
> 
> the directory they choose could have any name, what would you recommend? I could just leave out the bit about choosing a directory

Since the example demonstrates `~/Code` and then cloning I would prepend that directory so that the last command is `~/Code/sway/...`

**Comment by eureka-cpu** [comment]
> > > Only thing that I would change is that the contributing installation first tells the reader to change to any directory they want and then clone the repo into that directory however the last line, which builds, points at the repo at the top of their path rather than inside the directory which they have moved to. If they follow the instructions directly then they'll get an error
> > 
> > 
> > the directory they choose could have any name, what would you recommend? I could just leave out the bit about choosing a directory
> 
> Since the example demonstrates `~/Code` and then cloning I would prepend that directory so that the last command is `~/Code/sway/...`

I think I have a solution, instead of specifying a directory I think it may be best to just exclude it in general and leave it up to the user. I can also separate the build and move commands into separate steps so that the user can see that more clearly.

**Review by sezna — APPROVED** [review]

**Pull request merged**
